spec/requests/providers/providers_spec.rb

```diff
@@ -19,5 +19,6 @@
 
   it 'displays provider data' do
     expect(response.body).to include(provider.username)
+    expect(response.body).to include(provider.roles.gsub(/,(?![ ])/, ', '))
```

Comment:
I don't think it's a good idea to test something with exactly the same code as being implemented. I'd suggest:
```rspec
context 'with unspaced roles' do
  let(:provider) { create :provider, roles: 'foo,bar' }
  
  it 'should display roles with spaces do
    expect(response.body).to include('foo, bar')
  end
end
```